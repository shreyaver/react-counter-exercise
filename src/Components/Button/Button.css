.button {
    text-align: center;
    background: green;
    margin: 10px;
}