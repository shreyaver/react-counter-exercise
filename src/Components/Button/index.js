import React, { Component } from 'react';
import './Button.css';

// class Button extends Component {
//   render() {
//     const buttonStyles = {
//       color: '#fff'
//     }
//     return (<button type="button" className="button" style={buttonStyles} onClick={this.props.onClick}>{this.props.caption}</button>)
//   }
// }
const Button = (props) => {
  const buttonStyles = {
    color: '#fff'
  }
  return (
    <button type="button" className="button" style={buttonStyles} onClick={props.onClick}>
      {props.caption}
    </button>
  );
}
export default Button;