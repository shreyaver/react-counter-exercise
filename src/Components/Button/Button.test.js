import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';
import { shallow } from 'enzyme'

describe('button', () => {
  it('renders without crashing', () => {
    
    const button = renderer.create(<Button caption = "Click Me" />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    
    const wrapper = shallow(<Button caption = "Click Me" onClick={jest.fn()}/>);
    wrapper.find('button').simulate('click');
    expect(wrapper.props().onClick).toHaveBeenCalled();
  });
});
