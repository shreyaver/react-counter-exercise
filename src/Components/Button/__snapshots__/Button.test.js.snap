// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders without crashing 1`] = `
<button
  className="button"
  style={
    Object {
      "color": "#fff",
    }
  }
  type="button"
>
  Click Me
</button>
`;
